core/geometry.cpp

```diff
@@ -0,0 +1,62 @@
+#include "geometry.h"
+
+Geometry::Geometry(QObject *parent)
```

Comment:
Considering that this will only contain static functions, I think that Geometry should be a namespace only (not an object) and everything should be in the header file, no cpp.

Reply:
This would also make changes to the CMakeLists.txt unnecessary.